jgiven-spring/src/main/java/com/tngtech/jgiven/integration/spring/InternalSpringScenarioTest.java

```diff
@@ -0,0 +1,38 @@
+package com.tngtech.jgiven.integration.spring;
+
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.test.context.junit4.rules.SpringClassRule;
+import org.springframework.test.context.junit4.rules.SpringMethodRule;
+
+import com.tngtech.jgiven.base.ScenarioTestBase;
+import com.tngtech.jgiven.impl.ScenarioExecutor;
+
+/**
+ * Internal class necessary in order to provide the correct ordering of the {@link org.junit.rules.MethodRule}s. Must be public because of
+ * {@link SpringMethodRule}s validations, but should not be used directly.
+ * Instead, use {@link SpringScenarioTest}.
+ *
+ * @param <GIVEN>
+ * @param <WHEN>
+ * @param <THEN>
+ *
+ * @since 0.14.0
+ */
+public abstract class InternalSpringScenarioTest<GIVEN, WHEN, THEN> extends ScenarioTestBase<GIVEN, WHEN, THEN> implements BeanFactoryAware {
```

Comment:
no I think that would be overkill. The class is visible in any case, so just leave it as it is